hedy.py

```diff
@@ -587,19 +589,20 @@ def save_type_to_lookup(self, name, inferred_type):
     #  lookup entry is used to infer the type and continue the started validation. This approach might cause issues
     #  in case of cyclic references, e.g. b is b + 1. The flag `inferring` is used as a guard against these cases.
     def try_get_type_from_lookup(self, name):
-        # TODO: we should not just take the first match, take into account var reassignments
-        matches = [entry for entry in self.lookup if entry.name == name]
+        matches = [entry for entry in self.lookup if entry.name == hash_var(name)]
         if matches:
-            if not matches[0].inferred:
-                if matches[0].inferring:
-                    # TODO we know at this point there is a cyclic var reference, e.g. b = b + 1, for now do not raise
-                    matches[0].inferring = False
-                    matches[0].type_ = HedyType.any
-                    matches[0].inferred = True
+            # TODO: we should not just take the first match, take into account var reassignments
+            match = matches[0]
+            if not match.type_:
+                if match.inferring:  # there is a cyclic var reference, e.g. b = b + 1
```

Comment:
The comments combined with the name are really clear, but I am not sure `inferring` i a good name for the flag (without looking at the comment). I think it means something like "is this the var we are currently inferring"? So maybe `currently_inferring` or `is_currently_inferred` is better?

Reply:
The inferred flag is gone and the inferring is changed to currently_inferring